### DISCUSSION TITLE
"Warped" typo/mis-translation

### DISCUSSION THREAD

**Discussion by TestPilot02** [godotengine/godot-proposals#7094] [Editor]
In the UI settings I noticed that all of the settings related to "wrapping" are labled as "warping" or "warped" - I know english is not the first language of the Devs so I assume this is a simple mistranslation. 

In english to "warp" means to distort or skew the image, like stretching or smushing it in various ways. This is common terminology, examples can be seen in photoshop's "warp" tool, which applies a lattice and allows you to distort the image.

Wrapping, on the other hand, is when the mouse continues once it reaches the edge of the screen and keeps scrolling despite reaching the edge. I believe this was the intended implementation.

When I first encountered these settings I was extremely confused due to the mismatch in terms, I hope catching this spelling error early helps in the future of Godot! :)

![Screenshot 2023-06-15 193555](https://github.com/godotengine/godot-proposals/assets/114793822/b6e4f6c8-a2dd-4ffe-abd4-a8de30e1e6ff)

To be clear the fix is "Wrap" and "Wrapping"

**Comment by Calinou** [comment]
Renaming an editor setting breaks the setting for existing users (if they've changed it from the default value), but I guess it's acceptable for a minor release. That said, since 4.1 is in feature freeze, this likely has to wait for 4.2.

That said, [the method is internally called `Input.warp_mouse()`](https://github.com/godotengine/godot/blob/c0d8d91b15597209a1ca865c6ff4f84f06f7de86/core/input/input.cpp#L797-L799), not `Input.wrap_mouse()`. We definitely can't rename that one as it would break existing projects that use the method.

**Comment by theraot** [comment]
In gaming warp means teleporting: https://en.wikipedia.org/wiki/Warp_(video_games) and the mouse teleports from one end to the other. Meanwhile wrap is about enveloping, like the mouse going from one end to the other. It is fine. This is fine. *Insert this is fine meme.*

**Reply by TestPilot02** [reply]
Yes warp does mean teleporting which is why it's wrong. The mouse isn't teleporting...the setting is functionally wrapping the mouse from the left side of the screen to the right side of the screen (and vice versa) like when you see pacman go off one edge and appear on the other. That is a wrap... not a warp.

Again, wrapping is a common feature in other programs, this is a clear typo that has gotten baked in over time.

Here is another example:
https://www.youtube.com/watch?v=ppfvVW8Zqu4

the user is setting up the mouse to go from the left edge to the right edge of the screen. When you want to scroll past the edge of the screen but can't, you need to make the mouse "WRAP" to the other side of the screen.

**Reply by dalexeev** [reply]
> The mouse isn't teleporting...the setting is functionally wrapping the mouse from the left side of the screen to the right side of the screen (and vice versa) like when you see pacman go off one edge and appear on the other. That is a wrap... not a warp.

English is not my native language, so I cannot judge the use of these words linguistically. But mathematically it depends on whether you consider the screen to be topologically closed. If yes, then the cursor wraps, if not, then it warps.

**Reply by TestPilot02** [reply]
Fair enough, my argument is in favor of typical usage surrounding this - specifically Blender being the penultimate example considering it is a likely software many (if not most) users of Godot will be using. Also your use of English is far too advanced for you to continue claiming you can't analyze a word's value linguistically. You understand every single word better than most native speakers (clearly!) :)

Another example I thought of: Word wrapping within text documents. We don't call it "word warping"

**Reply by AThousandShips** [reply]
> Word wrapping

Not really the same thing, wrapping has an analogue, warp does not 

> Also your use of English is far too advanced for you to continue claiming you can't analyze a word's value linguistically

I mean doesn't really follow, I'm not a native English speaker, but I'm very eloquent, but I don't always understand the complexity and details of English words because I have a limited experience with them in all their native usage

**Reply by AThousandShips** [reply]
Now I agree that it might be a bit confusing, but it's technically correct (the best kind of correct, as we all know), and as mentioned above, breaking comparability here is not acceptable, especially if the reason is mild confusion or not being exactly correct 

What we could do is add details to the tooltip to mention wrapping too, like "warps, or wraps"

**Reply by dalexeev** [reply]
> Another example I thought of: Word wrapping within text documents. We don't call it "word warping"

In this case, the beginning of each line is considered a continuation of the previous one, and the entire text is mentally considered to be one continuous line.

I agree that the difference is very subtle and depends on how we look at the thing. For example here the author uses the word "wrap": godotengine/godot#78033. And in general "wrap" is used more often than "warp" in the API. The exceptions are the mouse cursor API and the `FastNoiseLite` class.

In the case of the `warp_mouse()` method, this is definitely correct, as it teleports the cursor to the specified position. In the case of the `FEATURE_MOUSE_WARP` constant, I tend to agree with you (omitting my comment about topology):

> FEATURE_MOUSE_WARP = 4
> Display server supports warping mouse coordinates to keep the mouse cursor constrained within an area, but looping when one of the edges is reached. Windows, macOS, Linux (X11)

But renaming is prohibited in minor and patch releases, so the only thing we can do is improve the documentation.
